lib/openstudio-standards/weather/information.rb

```diff
@@ -362,6 +362,80 @@ def self.get_climate_zone_from_degree_days(hdd18, cdd10)
+
+      # extraterrestrial normal irradiance, W/m^2
+      extraterrestrial_normal_irradiance_degrees = 360 * (day_of_year - 3) / 365.0
+      extraterrestrial_normal_irradiance_radians = extraterrestrial_normal_irradiance_degrees * (Math::PI / 180.0)
+      extraterrestrial_normal_irradiance = 1367.0 * (1.0 + 0.033 * Math.cos(extraterrestrial_normal_irradiance_radians))
+
+      # declination
+      day_angle_degrees = 360.0 * (day_of_year + 284) / 365.0
+      day_angle_radians = day_angle_degrees * (Math::PI / 180.0)
+      declination_degrees = 23.45 * Math.sin(day_angle_radians)
+      declination_radians = declination_degrees * (Math::PI / 180.0)
+
+      # air mass exponents from optical depth
+      tau_b = design_day.ashraeClearSkyOpticalDepthForBeamIrradiance
+      tau_d = design_day.ashraeClearSkyOpticalDepthForDiffuseIrradiance
+      ab = 1.454 - (0.406 * tau_b) - (0.268 * tau_d) + (0.021 * tau_b * tau_d)
+      ad = 0.507 + (0.205 * tau_b) - (0.080 * tau_d) - (0.190 * tau_b * tau_d)
+
+      global_irradiance_array = []
+      (0..23).to_a.each do |local_standard_time_hour|
+        # apparent solar time
+        apparent_solar_time = local_standard_time_hour + (equation_of_time_minutes / 60.0) + (site_longitude_degrees - time_zone_longitude_degrees)/15.0
+
+        # hour angle
+        hour_angle_degrees = 15.0 * (apparent_solar_time - 12.0)
+        hour_angle_radians = hour_angle_degrees * (Math::PI / 180.0)
+
+        # solar altitude
+        solar_altitude_radians = Math.asin(Math.cos(site_latitude_radians) * Math.cos(declination_radians) * Math.cos(hour_angle_radians) + Math.sin(site_latitude_radians) * Math.sin(declination_radians))
+        solar_altitude_degrees = solar_altitude_radians * (180.0 / Math::PI)
+
+        # equation 16 air mass
+        # equation 17 and 18 irradiance calculation
+        if solar_altitude_degrees > 0
+          air_mass = 1 / (Math.sin(solar_altitude_radians) + 0.50572 * (6.07995 + solar_altitude_degrees)**(-1.6364))
+          beam_normal_irradiance = extraterrestrial_normal_irradiance * Math.exp(-tau_b * air_mass**ab)
+          diffuse_horizontal_irradiance = extraterrestrial_normal_irradiance * Math.exp(-tau_d * air_mass**ad)
+        else
+          air_mass = nil
+          beam_normal_irradiance = 0.0
+          diffuse_horizontal_irradiance = 0.0
+        end
+        global_irradiance = beam_normal_irradiance + diffuse_horizontal_irradiance
+        global_irradiance_array << global_irradiance
+
+        # puts "For local_standard_time_hour #{local_standard_time_hour}, apparent_solar_time #{apparent_solar_time}, hour_angle_degrees #{hour_angle_degrees}, solar_altitude_degrees #{solar_altitude_degrees}, air_mass #{air_mass}, beam_normal_irradiance #{beam_normal_irradiance}, diffuse_horizontal_irradiance #{diffuse_horizontal_irradiance}"
+      end
+
+      average_daily_global_irradiance = global_irradiance_array.sum / 24.0
```

Comment:
I noticed you are averaging over 24 hours of the day. I am still not sure what PHIUS meant by that…

Reply:
Yes, I went with the daily average. It's both a bit more straightforward to calculate, and gives numbers closer to the range in Graham Wright's paper. Only during sun hours is both more complicated, and means that the average value across design days isn't comparable, because the sun hours aren't the same.